gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -388,6 +394,37 @@ public GsonBuilder setObjectToNumberStrategy(ToNumberStrategy objectToNumberStra
     return this;
   }
 
+  /**
+   * Configures Gson to apply a specific missing field value strategy during deserialization.
+   * The strategy is used during reflection-based deserialization when the JSON data does
+   * not contain a value for a field. A field with explicit JSON null is not considered missing.
+   *
+   * @param missingFieldValueStrategy strategy handling missing field values
+   * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
+   * @see MissingFieldValueStrategy#DO_NOTHING The default missing field value strategy
+   * @since $next-version$
+   */
+  public GsonBuilder setMissingFieldValueStrategy(MissingFieldValueStrategy missingFieldValueStrategy) {
+    this.missingFieldValueStrategy = Objects.requireNonNull(missingFieldValueStrategy);
+    return this;
+  }
+
+  /**
+   * Configures Gson to apply a specific unknown field strategy during deserialization.
+   * The strategy is used during reflection-based deserialization when an unknown field
+   * is encountered in the JSON data. If a field which is excluded from deserialization
+   * appears in the JSON data it is considered unknown as well.
```

Comment:
There might be use cases where a field excluded from deserialization should not be considered unknown, but should just be silently ignored.

With the current `UnknownFieldStrategy` API users might be able to determine themselves if a field is excluded, but not in a very reliable way.

We could possibly add a `boolean excluded` parameter to `UnknownFieldStrategy.handleUnknownField` whose value is `true` if `ReflectiveTypeAdapterFactory` finds a `BoundField` but it has `!field.deserialized`. Though I am not sure if that is really worth it. What are your opinions?